src/main/java/org/dave/compactmachines3/block/BlockMachine.java

```diff
@@ -227,15 +228,20 @@ public static void spawnItemWithNBT(World world, BlockPos pos, EnumMachineSize s
         }
 
         ItemStack stack = new ItemStack(Blockss.machine, 1, size.getMeta());
-        NBTTagCompound compound = new NBTTagCompound();
-        compound.setInteger("coords", te.coords);
-        if(te.hasOwner()) {
-            compound.setUniqueId("owner", te.getOwner());
-        }
-        if(te.hasNewSchema()) {
-            compound.setString("schema", te.getSchemaName());
+        // If a player places a machine, doesn't use it, and picks it back up, they can't stack it as it has NBT data.
+        if (ConfigurationHandler.MachineSettings.allowPickupEmptyMachines && te.id == -1 && !te.hasNewSchema()) {
+            // Don't add NBT data if the config value is set to true and the machine id is -1 and it doesn't have a schema
+        } else {
+            NBTTagCompound compound = new NBTTagCompound();
+            compound.setInteger("machineId", te.id);
+            if (te.hasOwner()) {
+                compound.setUniqueId("owner", te.getOwner());
+            }
+            if (te.hasNewSchema()) {
+                compound.setString("schema", te.getSchemaName());
+            }
+            stack.setTagCompound(compound);
```

Comment:
Possibly breaking change, did this config value exist before, or was it not documented?

Reply:
It did not exist before. I added it, but it defaults to false so as to explicitly not cause breaking changes. I refactored the config a bit, mainly adding more language key translations and some refactoring. You can find these changes in the `ConfigurationHandler` class I believe.